### DISCUSSION TITLE
Portalverbund - Darstellungsfilter reagiert unerwartet

### DISCUSSION THREAD

**Discussion by faulgurke** [e-netze/webgis-community#96] [Q&A]
In einer Portalverbund-Anwendung gibt es folgenden Darstellungsfilter:
Betroffene Layer: 
Ersichtlichmachung_Sachbearbeiter;Fangebene;FS_Sachbearbeiter;Linien_Sachbearbeiter;Verkehrsfl_Sachbearbeiter;...
SQL Filter:
[role-parameter:VKZ,gem_nr>{0}.]
Anzeigetyp:
locked

Erwartet wurde die **Nicht-Darstellung** der Layer, wenn der Rollenparameter (des Users) nicht dem Filter entspricht. In diesem Fall hat der Benutzer keinen Rollenparameter "VKZ".

Allerdings werden die Layer dargestellt und es erscheint (pro Layer) eine Fehlermeldung:

![fehlermeldung_darstellungsfilter](https://github.com/gis-eni/webgis-forum/assets/100088983/6b846610-e251-4d68-b416-1f4b818d5653)

Tipps?

**Comment by jugstalt** [comment]
Die Syntax für die Filter ist:

`` [role-parameter:parameter_name,filter(,logical_operator)]``

zB:

``[role-parameter:gemnr,GEM_NR like '{0}%]``

bzw wenn es mehrer GEM_NR Parameter gibt:

``[role-parameter:gemnr,GEM_NR like '{0}%',OR]``

Hier werden die einzelnen Werte mit ``OR`` verknüpft

Den Filter von oben versteh ich nicht :/ Und was wenn der Anwender gar keinen VKZ Rollenparameter hat ergibt sich ``gem_nr>.`` ??

**Reply by faulgurke** [reply]
Die Syntax der Filter sollte passen, das ist ja schon jahrelang im WebGIS4 gelaufen. Wenn ein Anwender keinen Rollenparameter VKZ hat, wird der Inhalt nicht dargestellt. Es gibt zwei Gruppen von Anwendern, die über unterschiedliche Filter auf den Datenbestand schauen.
Die eine Gruppe sieht nur "ihre" Filterung, die andere das Ergebnis des anderen Filters.
Und das ist das Problem welches ich oben beschrieben habe.

**Comment by gis-eni** [comment]
Ich versteh nicht, wie das jemals funktioniert hat. Das Statement ``gem_nr>.`` macht für mich keinen Sinn.
Auch wenn ein VKZ vorhanden ist: ``gem_nr>11111.`` dürfte ja auch nicht funktionieren.

Die Logik für diesen Filter hat sich im webgis5 nicht geändert
